conans/model/requires.py

```diff
@@ -289,6 +289,8 @@ def transform_downstream(self, pkg_type, require, dep_pkg_type):
                 downstream_require = Requirement(require.ref, headers=False, libs=False, run=require.run)
             elif pkg_type is PackageType.HEADER:
                 downstream_require = Requirement(require.ref, headers=require.headers, libs=require.libs, run=require.run)
+            elif pkg_type is PackageType.BUILD_SCRIPTS:
+                raise ConanError("build-scripts package must be used as a tool_requires(), not a regular requirement")
```

Comment:
This check is about a build-script depending on a library, not really about it being used as regular ``requires``. This might be an extra check to do, but it seems we want to do a general check over build-scripts being used only as ``tool-requires``